monai/apps/deepgrow/transforms.py

```diff
@@ -0,0 +1,623 @@
+        self.ref_image = ref_image
+        self.slice_only = slice_only
+        self.channel_first = channel_first
+        self.mode = ensure_tuple_rep(mode, len(self.keys))
+        self.align_corners = ensure_tuple_rep(align_corners, len(self.keys))
+        self.meta_key_postfix = meta_key_postfix
+        self.start_coord_key = start_coord_key
+        self.end_coord_key = end_coord_key
+        self.original_shape_key = original_shape_key
+        self.cropped_shape_key = cropped_shape_key
+
+    def __call__(self, data):
+        meta_dict = data[f"{self.ref_image}_{self.meta_key_postfix}"]
+
+        for idx, key in enumerate(self.keys):
+            image = data[key]
+
+            # Undo Resize
+            current_size = image.shape
+            cropped_size = meta_dict[self.cropped_shape_key]
+            if np.any(np.not_equal(current_size, cropped_size)):
+                resizer = Resize(spatial_size=cropped_size[1:], mode=self.mode[idx])
+                image = resizer(image, mode=self.mode[idx], align_corners=self.align_corners[idx])
+
+            # Undo Crop
+            original_shape = meta_dict[self.original_shape_key]
+            result = np.zeros(original_shape, dtype=np.float32)
+            box_start = meta_dict[self.start_coord_key]
+            box_end = meta_dict[self.end_coord_key]
+
+            sd = min(len(box_start), len(box_end), len(image.shape[1:]))  # spatial dims
+            slices = [slice(None)] + [slice(s, e) for s, e in zip(box_start[:sd], box_end[:sd])]
+            slices = tuple(slices)
+            result[slices] = image
+
+            # Undo Spacing
+            current_size = result.shape[1:]
+            spatial_shape = np.roll(meta_dict["spatial_shape"], 1).tolist()
+            spatial_size = spatial_shape[-len(current_size) :]
+
+            if np.any(np.not_equal(current_size, spatial_size)):
+                resizer = Resize(spatial_size=spatial_size, mode=self.mode[idx])
+                result = resizer(result, mode=self.mode[idx], align_corners=self.align_corners[idx])
+
+            # Undo Slicing
+            slice_idx = meta_dict.get("slice_idx")
+            if slice_idx is None or self.slice_only:
+                final_result = result if len(result.shape) <= 3 else result[0]
+            else:
+                slice_idx = meta_dict["slice_idx"][0]
+                final_result = np.zeros(spatial_shape)
+                if self.channel_first:
+                    final_result[slice_idx] = result
+                else:
+                    final_result[..., slice_idx] = result
+            data[key] = final_result
+
+            meta = data.get(f"{key}_{self.meta_key_postfix}")
+            if meta is None:
+                meta = dict()
+                data[f"{key}_{self.meta_key_postfix}"] = meta
+            meta["slice_idx"] = slice_idx
+            meta["affine"] = meta_dict["original_affine"]
+        return data
+
+
+class AddGuidanceFromPointsd(Randomizable, Transform):
```

Comment:
should be just ``MapTransform`` without Randomizable since there is no random state in this transform?

Reply:
user click is a random